addons/flannel/template/base/install.sh

```diff
@@ -67,6 +67,16 @@ function flannel() {
         kubectl -n kube-flannel apply -k "$dst/"
     fi
 
+    # We will remove the flannel pods to let it be re-created
+    # in order to workaround the issue scenario described in
+    # https://github.com/flannel-io/flannel/issues/1721
+    if [ "$KUBERNETES_UPGRADE" == "1" ]; then
+       sleep 30
```

Comment:
i have an issue with timeouts of arbitrary length. it introduces flakes in our product. could we get creative here?

Reply:
The sleep if just to give a sometime to ensure that all was applied (line 67 ` kubectl -n kube-flannel apply -k "$dst/"`), not sure how creative we would be.

Reply:
could we not check that the flannel ds pods are running and ready?

Reply:
I think we can just move the test bellow the health check. That is a good point thank you.